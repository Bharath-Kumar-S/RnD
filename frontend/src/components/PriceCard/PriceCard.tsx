type PriceCardProps = {
  currency: string;
  price: number;
};

export const PriceCard = ({ currency, price }: PriceCardProps) => {
  return (
    <h3 className="font-bold">
      {currency}: <span className="font-extrabold">{price}</span>
    </h3>
  );
};
