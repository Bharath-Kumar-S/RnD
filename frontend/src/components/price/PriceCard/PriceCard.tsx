type PriceCardProps = {
  pair: string;
  price: number;
};

export const PriceCard = ({ pair, price }: PriceCardProps) => {
  return (
    <div className="bg-white/10 backdrop-blur-lg rounded-xl p-6 shadow-xl border border-white/20">
      <h2 className="text-xl font-semibold text-blue-200 mb-4">{pair}</h2>
      <div className="flex items-baseline justify-between">
        <span className="text-3xl font-bold text-white">
          {price ? (price % 1 === 0 ? price : price.toFixed(6)) : "..."}
        </span>
      </div>
    </div>
  );
};
