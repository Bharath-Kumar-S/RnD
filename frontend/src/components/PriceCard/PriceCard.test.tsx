// src/components/PriceCard.test.tsx
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { PriceCard } from "./PriceCard";

describe("PriceCard", () => {
  it("renders the currency and price correctly", () => {
    render(<PriceCard currency="USD" price={100} />);

    expect(screen.getByText("USD:")).toBeInTheDocument();
    expect(screen.getByText("100")).toBeInTheDocument();
  });

  it("applies the correct font styles", () => {
    render(<PriceCard currency="EUR" price={200} />);

    const currencyElement = screen.getByText("EUR:");
    const priceElement = screen.getByText("200");

    expect(currencyElement).toHaveClass("font-bold");
    expect(priceElement).toHaveClass("font-extrabold");
  });

  it("should match snapshots", () => {
    const { container } = render(<PriceCard currency="GBP" price={300} />);
    expect(container).toMatchSnapshot();
  });
});
