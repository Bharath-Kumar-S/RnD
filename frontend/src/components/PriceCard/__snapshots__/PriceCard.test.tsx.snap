// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PriceCard > should match snapshots 1`] = `
<div>
  <h3
    class="font-bold"
  >
    GBP
    : 
    <span
      class="font-extrabold"
    >
      300
    </span>
  </h3>
</div>
`;
